### ISSUE TITLE
VS for Mac: "CopyGeneratedJavaResourceClasses" task was not given a value for the required parameter "SourceTopDirectory". MSB4044

### ISSUE THREAD

**Issue by inimirpaz** [xamarin/xamarin-android#4370]
Related to #4068, since that issue is closed but many still faces it (VS for Mac):
building the app requires the user to build twice, because the first time the 

"CopyGeneratedJavaResourceClasses" task was not given a value for the required parameter "SourceTopDirectory". MSB4044

error is given.

Visual Studio Professional 2019 for Mac
Version 8.4.8 (build 2)

Roslyn (Language Service)
3.4.0-beta4-19562-05+ff930dec4565e2bc424ad3bf3e22ecb20542c87d

Xamarin.Android
Versione: 10.1.3.7 (Visual Studio Professional)
Commit: xamarin-android/d16-4/d66aed0

Xamarin.Forms 4.5.0.356

**Comment by radekdoulik** [comment]
Could you please attach diagnostic build output?

**Comment by inimirpaz** [comment]
> Could you please attach diagnostic build output?

Please find the log attached
[VSlog.txt.zip](https://github.com/xamarin/xamarin-android/files/4307476/VSlog.txt.zip)

**Comment by brendanzagaeski** [comment]
Thanks for the build log!

This log closely matches the behavior described in <https://github.com/xamarin/xamarin-android/issues/4068#issuecomment-576493386>. In particular, the project is using the Xamarin.AndroidX.Migration NuGet package and the log shows:

```
Skipping target "_PrepareCreateBaseApk" because all output files are up-to-date with respect to the input files.
```

As a consequence, the `AaptTemporaryDirectory` property is left blank, which causes an error later in the `CopyGeneratedJavaResourceClasses` task.

Based on that information, I'll tentatively close this as a duplicate of <https://github.com/xamarin/xamarin-android/issues/4068>.

**Optional follow-up steps:**

The fix for <https://github.com/xamarin/xamarin-android/issues/4068> is now available in a Preview version on macOS. If you get a chance, you can check that the new preview version resolves the issue for your projects too.

Fix included in Xamarin.Android 10.2.0.100.

Fix included on macOS in Visual Studio 2019 for Mac version 8.5 Preview 6. To try the Preview version that includes the fix, check for the latest updates on the **Preview** [updater channel](https://docs.microsoft.com/visualstudio/mac/update).

If that version does *not* resolve the issue, please do reply back with a comment to let the team know, and the team will reopen this issue for further investigation. Thanks!

**Comment by brendanzagaeski** [comment]
Duplicate of #4068

**Comment by inimirpaz** [comment]
> Thanks for the build log!
> 
> This log closely matches the behavior described in [#4068 (comment)](https://github.com/xamarin/xamarin-android/issues/4068#issuecomment-576493386). In particular, the project is using the Xamarin.AndroidX.Migration NuGet package and the log shows:
> 
> ```
> Skipping target "_PrepareCreateBaseApk" because all output files are up-to-date with respect to the input files.
> ```
> 
> As a consequence, the `AaptTemporaryDirectory` property is left blank, which causes an error later in the `CopyGeneratedJavaResourceClasses` task.
> 
> Based on that information, I'll tentatively close this as a duplicate of #4068.
> 
> **Optional follow-up steps:**
> 
> The fix for #4068 is now available in a Preview version on macOS. If you get a chance, you can check that the new preview version resolves the issue for your projects too.
> 
> Fix included in Xamarin.Android 10.2.0.100.
> 
> Fix included on macOS in Visual Studio 2019 for Mac version 8.5 Preview 6. To try the Preview version that includes the fix, check for the latest updates on the **Preview** [updater channel](https://docs.microsoft.com/visualstudio/mac/update).
> 
> If that version does _not_ resolve the issue, please do reply back with a comment to let the team know, and the team will reopen this issue for further investigation. Thanks!

Thank you,

is there any scheduled release date for VS 2019 for Mac v. 8.5 yet? We are trying to stay out of the preview channel if possibile, but if it is still a long wait we'd consider to move onto the preview channel.
Thanks.

**Comment by brendanzagaeski** [comment]
As a quick cross-reference on this item, Visual Studio 2019 for Mac version 8.5 was published today: https://github.com/xamarin/xamarin-android/issues/4068#issuecomment-602991958
